packages/core/src/transaction.ts

```diff
@@ -22,12 +22,12 @@ export class Transaction {
   readonly #afterRollbackHooks: Set<TransactionCallback> = new Set();
   readonly #afterHooks: Set<TransactionCallback> = new Set();
 
-  private readonly savepoints: Transaction[] = [];
+  private readonly savepoints: Map<string, Transaction> = new Map();
```

Comment:
prefer `#private`

Reply:
Done as well as the other internal methods